src/Dapr.Client/ServiceInvocationException.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+namespace Dapr.Client
+{
+    using Dapr.Client.Http;
+    using System;
+    using System.Net;
+
+    /// <summary>
+    /// This class is only needed if the app you are calling app is listening on gRPC.
+    /// It contains propertes that represent status info that may be populated for an gRPC response.
+    /// </summary>
+    public class ServiceInvocationException<TRequest, TResponse> : Exception
+    {
+        /// <summary>
+        /// The constructor.
+        /// </summary>
+        public ServiceInvocationException(ServiceInvocationResponse<TRequest, TResponse> response)
+        {
+            this.Response = response;
+        }
+
+        /// <summary>
+        /// The constructor.
+        /// </summary>
+        public ServiceInvocationException(string message, ServiceInvocationResponse<TRequest, TResponse> response)
+            : base(message)
+        {
+            this.Response = response;
+        }
+
+        /// <summary>
+        /// The constructor.
+        /// </summary>
+        public ServiceInvocationException(string message, Exception innerException, ServiceInvocationResponse<TRequest, TResponse> response)
+            : base(message, innerException)
+        {
+            this.Response = response;
+        }
+
+
+        /// <summary>
+        /// The gRPC Error Message
+        /// </summary>
+        public string GrpcErrorMessage { get; set; }
```

Comment:
What's the purpose of this? how's it different from the message of the exception? Is this the message returned byt he dapr sidecar? example?

Reply:
yes, that's the message returned by Dapr side car. https://github.com/dapr/dapr/blob/6af5d6e6a497aae2cf9deb4759f7be3be20193cf/pkg/messaging/v1/util.go#L319

and GrpcStatusCode is the grpc status code returned by Dapr sidecar which is a mapped value from the http status code.